components/ei-analytics/org.wso2.analytics.solutions.ei.analytics/widgets/EIAnalyticsSearchBox/src/EIAnalyticsSearchBox.jsx

```diff
@@ -16,24 +16,205 @@
  * under the License.
  */
 
+import React from 'react';
+import PropTypes from 'prop-types';
 import Widget from '@wso2-dashboards/widget';
+import {MuiThemeProvider, createMuiTheme} from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+import MenuItem from '@material-ui/core/MenuItem';
+import Chip from '@material-ui/core/Chip';
+import Typography from '@material-ui/core/Typography';
+import Popper from '@material-ui/core/Popper';
+import Paper from '@material-ui/core/Paper';
 import Select from 'react-select';
-import './react-select.css';
+import JssProvider from 'react-jss/lib/JssProvider';
+import {Scrollbars} from 'react-custom-scrollbars';
 
-class EIAnalyticsSearchBox extends Widget {
+const darkTheme = createMuiTheme({
+    palette: {
+        type: 'dark'
+    }
+});
+
+const lightTheme = createMuiTheme({
+    palette: {
+        type: 'light'
+    }
+});
+
+let openPopper = false;
+
+function NoOptionsMessage(props) {
```

Comment:
Why these functions resides outside a class?

Reply:
That's how material ui implements autocomplete using react select.